flake.nix

```diff
@@ -0,0 +1,37 @@
+{
+  description = "Compio dev shell";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    rust-overlay,
+    flake-utils,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (
+      system: let
+        overlays = [(import rust-overlay)];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+      in
+        with pkgs; {
+          devShells.default = mkShell {
+            buildInputs = [
+              openssl
+              pkg-config
+              (rust-bin.selectLatestNightlyWith (toolchain:
+                toolchain.default.override {
+                  extensions = ["rust-src"];
```

Comment:
do we need it?

Reply:
Yes. Without this rust analyzer won't work properly.